apps/app/components/LectureForm/index.js

```diff
@@ -0,0 +1,191 @@
+              index === self.findIndex((t) => t.id === ninja.id)
+          );
+        });
+      });
+    });
+  }, [events]);
+
+  const onFinish = (values) => {
+    if (id) {
+      api
+        .createLecture(id, values)
+        .then(() => {
+          router.push("/admin/lectures");
+        })
+        .catch((error) => notification["error"](error.data?.errors));
+    } else {
+      api
+        .createLecture(values)
+        .then((response) => {
+          router.push("/admin/lectures");
+        })
+        .catch((error) => notification["error"](error.data?.errors));
+    }
+  };
+
+  return (
+    <>
+      <Row justify="space-between">
+        <Title level={2}>
+          {id && ninjas[0]
+            ? ninjas[0].first_name + " " + ninjas[0].last_name
+            : "Nova Sessão"}
+        </Title>
+        <Space>
+          <Link href="/admin/lectures">
+            <Button
+              danger
+              shape="circle"
+              size="large"
+              icon={<CloseOutlined />}
+            />
+          </Link>
+          <Button
+            shape="circle"
+            type="primary"
+            size="large"
+            icon={<SaveOutlined />}
+            onClick={() => form.submit()}
+          />
+        </Space>
+      </Row>
+      <Row align="middle">
+        <Col xs={24} sm={24} md={20} lg={16} xl={12}>
+          <Form
+            {...{
+              labelCol: { span: 8 },
+              wrapperCol: { span: 16 },
+            }}
+            form={form}
+            onFinish={onFinish}
+          >
+            <Form.Item
+              label="Evento"
+              name="lecture[event_id]"
+              rules={[{ required: true }]}
+            >
+              <Select
+                placeholder="Escolha um evento"
+                style={{ width: "50%" }}
+                onChange={setSelectedEvent}
+                value={
+                  Object.keys(selectedEvent).length === 0
+                    ? undefined
+                    : selectedEvent
+                }
+              >
+                {events.map((event) => (
+                  <Select.Option key={event.id} value={event.id}>
+                    <div>
+                      {`${event.title} - ${event.start_time} - ${event.end_time}`}
+                    </div>
+                  </Select.Option>
+                ))}
+              </Select>
+            </Form.Item>
+
+            <Form.Item
+              label="Ninja"
+              name="lecture[ninja_id]"
+              rules={[{ required: true }]}
+            >
+              <Select
+                placeholder="Escolha pelo menos um Ninja"
+                style={{ width: "50%" }}
+                onChange={setSelectedNinja}
+                value={
+                  Object.keys(selectedNinja).length === 0
+                    ? undefined
+                    : selectedNinja
+                }
+              >
+                {ninjas.map((ninja) => (
```

Comment:
Here, should only appear ninjas that doesn't already have a lecture for the current selected event. Otherwise, if I select a ninja and a mentor that already have a lecture I will receive an error from the API saying "já existe no sistema". Bearing that in mind, add the same constraint to the mentors listing.